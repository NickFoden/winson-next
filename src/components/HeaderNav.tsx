import React, { useState, useRef } from "react";
import Link from "next/link";
import { useOnClickOutside } from "../utils/clickOutside";
import Burger from "../components/Burger";
import Menu from "../components/Menu";

const HeaderNav = () => {
  const [open, setOpen] = useState(false);
  const node = useRef();

  useOnClickOutside(node, () => setOpen(false));
  return (
    <>
      <nav ref={node}>
        <Link href="/">
          <a className="home">Win Son</a>
        </Link>
        <Burger open={open} setOpen={setOpen} />
        <div className="links_container">
          <Link href="/restaurant">
            <a className="venue">Restaurant</a>
          </Link>
          <Link href="/bakery">
            <a className="venue">Bakery</a>
          </Link>
        </div>
      </nav>
      <Menu open={open} setOpen={setOpen} />
      <style jsx>{`
        nav {
          display: flex;
          justify-content: space-between;
          padding: 1rem;
          width: 100vw;
        }
        .home {
          color: black;
          font-family: "Courier New", Courier, monospace;
          text-decoration: none;
        }
        .home:hover {
          opacity: 0.7;
        }
        .links_container {
          display: none;
        }
        .venue {
          color: black;
          font-family: "Courier New", Courier, monospace;
          margin-right: 2rem;
          text-decoration: none;
        }
        .venue:hover {
          opacity: 0.7;
        }
        @media screen and (min-width: 500px) {
          .links_container {
            display: block;
          }
        }
      `}</style>
    </>
  );
};

export default HeaderNav;
