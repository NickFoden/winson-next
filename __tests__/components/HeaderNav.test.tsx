import { render } from "@testing-library/react";
import HeaderNav from "../../src/components/HeaderNav";

describe("HeaderNav", () => {
  test("HeaderNav Renders", () => {
    const { getByText, queryAllByText } = render(<HeaderNav />);
    expect(getByText("Win Son")).toBeTruthy();
    expect(queryAllByText("Restaurant")).toHaveLength(2);
    expect(queryAllByText("Bakery")).toHaveLength(2);
  });
});
