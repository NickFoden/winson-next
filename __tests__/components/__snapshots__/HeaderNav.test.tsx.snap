// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav HeaderNav Renders 1`] = `
<DocumentFragment>
  <nav
    class="jsx-110598338"
  >
    <a
      class="jsx-110598338"
      href="/"
    >
      <img
        class="jsx-110598338 logo"
        src="/svg/morewinsonbuttons_winson.svg"
      />
    </a>
    <div
      class="jsx-264131979 container"
    >
      <button
        aria-expanded="false"
        aria-label="Toggle menu"
        class="Burgerstyled__StyledBurger-n2mypy-0 dpBCVs"
      >
        <span
          class="jsx-264131979"
        />
        <span
          class="jsx-264131979"
        />
        <span
          class="jsx-264131979"
        />
      </button>
    </div>
    <div
      class="jsx-110598338 links_container"
    >
      <div
        class="jsx-110598338 links_container"
      >
        <a
          class="jsx-110598338 venue"
          href="/restaurant"
        >
          <img
            class="jsx-110598338 logo_links"
            src="/svg/morewinsonbuttons_restaurant.svg"
          />
        </a>
        <a
          class="jsx-110598338 venue"
          href="/bakery"
        >
          <img
            class="jsx-110598338 logo_links"
            src="/svg/morewinsonbuttons_bakery.svg"
          />
        </a>
      </div>
      <div
        class="jsx-110598338 links_container"
      />
    </div>
  </nav>
  <nav
    aria-hidden="true"
    class="Menustyled__StyledMenu-sc-1ipxm0t-0 ePfPiP"
  >
    <a
      class="jsx-2513438216"
      href="/restaurant"
    >
      <img
        class="jsx-2513438216 logo_links"
        src="/svg/morewinsonbuttons_restaurant.svg"
      />
    </a>
    <a
      class="jsx-2513438216"
      href="/restaurant/menu"
    >
      <img
        class="jsx-2513438216 sub_menu"
        src="/svg/menu.svg"
      />
    </a>
    <a
      class="jsx-2513438216 sub"
      href="/restaurant/order-now"
    >
      <img
        class="jsx-2513438216 sub_menu"
        src="/svg/ordernow.svg"
      />
    </a>
    <a
      class="jsx-2513438216"
      href="/bakery"
    >
      <img
        class="jsx-2513438216 logo_links"
        src="/svg/morewinsonbuttons_bakery.svg"
      />
    </a>
    <a
      class="jsx-2513438216"
      href="/bakery/menu"
    >
      <img
        class="jsx-2513438216 sub_menu"
        src="/svg/menu.svg"
      />
    </a>
    <a
      class="jsx-2513438216 sub"
      href="/bakery/order-now"
    >
      <img
        class="jsx-2513438216 sub_menu"
        src="/svg/ordernow.svg"
      />
    </a>
  </nav>
</DocumentFragment>
`;
